File: apps/web/src/lib/site.ts
export const locales = ["en", "zh"] as const;
export type Locale = (typeof locales)[number];
export const defaultLocale: Locale = "en";

const localePattern = new RegExp(`^/(${locales.join("|")})(?=/|$)`);

export function isLocale(value: string): value is Locale {
  return (locales as readonly string[]).includes(value);
}

/** Strip `/en` or `/zh` prefix; returns `/` for locale root. */
export function stripLocale(pathname: string): string {
  const stripped = pathname.replace(localePattern, "");
  return stripped === "" ? "/" : stripped;
}

export function withLocale(lang: Locale, path = "/"): string {
  const normalized = path.startsWith("/") ? path : `/${path}`;
  if (normalized === "/") return `/${lang}`;
  return `/${lang}${normalized}`;
}

/** Public site meta — no employers, school, or private email on the site */
export const site = {
  name: "Nathan Zhao",
  handle: "nathan",
  jobTitle: "Fullstack Engineer",
  /** Canonical production origin (no trailing slash) */
  url: "https://sundaydev.vercel.app",
  social: {
    github: "https://github.com/sundaydev-arch",
    website: "https://sundaydev.vercel.app/",
    /** Public Cal.com profile — override path via NEXT_PUBLIC_CAL_LINK */
    cal: "https://cal.com/nathan-zhao",
  },
  /** Topics for schema.org / GEO — capability-focused, no employers */
  knowsAbout: [
    "TypeScript",
    "Next.js",
    "React",
    "NestJS",
    "FastAPI",
    "Go",
    "multi-tenant portals",
    "BFF",
    "Open API",
  ],
} as const;

export type Project = {
  id: string;
  title: string;
  role: string;
  summary: string;
  highlights: string[];
  tags: string[];
  year: string;
  featured?: boolean;
};

File: apps/mobile/lib/core/site.dart
/// Public site meta — mirrors `apps/web/src/lib/site.ts`.
/// No employers, school, or private email on the public surface.
const locales = ["en", "zh"];
typedef LocaleCode = String;

const defaultLocale = "en";

bool isLocale(String value) => locales.contains(value);

class Site {
  const Site._();

  static const name = "Nathan Zhao";
  static const handle = "nathan";
  static const jobTitle = "Fullstack Engineer";

  /// Canonical production origin (no trailing slash).
  static const url = "https://sundaydev.vercel.app";

  static const github = "https://github.com/sundaydev-arch";
  static const website = "https://sundaydev.vercel.app/";
  static const cal = "https://cal.com/nathan-zhao";

  static const knowsAbout = [
    "TypeScript",
    "Next.js",
    "React",
    "NestJS",
    "FastAPI",
    "Go",
    "multi-tenant portals",
    "BFF",
    "Open API",
  ];
}

class Project {
  const Project({
    required this.id,
    required this.title,
    required this.role,
    required this.summary,
    required this.highlights,
    required this.tags,
    required this.year,
    this.featured = false,
  });

  final String id;
  final String title;
  final String role;
  final String summary;
  final List<String> highlights;
  final List<String> tags;
  final String year;
  final bool featured;

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      id: json["id"] as String,
      title: json["title"] as String,
      role: json["role"] as String,
      summary: json["summary"] as String,
      highlights: (json["highlights"] as List<dynamic>).cast<String>(),
      tags: (json["tags"] as List<dynamic>).cast<String>(),
      year: json["year"] as String,
      featured: json["featured"] as bool? ?? false,
    );
  }
}
